src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -512,13 +518,15 @@ int CreateRunRequest(int argc, char** argv, const char** env,
   req->add_command("\\.o-.*$");
   req->add_command("\\.git.*$");
   req->add_command("--environment_variables");
-  string env_vars;
+  string env_vars = "PWD=" + kPWDOverride + ",";
```

Comment:
Are you sure you want to do the override here? I think it's much cleaner to do this outside, as we used to, and just always call this with PWD set to /proc/self/cwd and in the whitelist. This /proc/self/cwd thing is very Android-specific. But it can be a separate CL to take it out, if you prefer.

Reply:
I thought about this. I actually think pwd is not android specific. We can't do a deterministic compilation without it, and relying on someone else setting it is unnecessary imo. I also think bazel does this by default for linux (https://github.com/bazelbuild/bazel/issues/1000#issuecomment-285344868)

Reply:
Also note that we always did it within rbecc: https://github.com/ola-rozenfeld/tools_remote/blob/tool/src/main/cc/proxy_client/proxy_client.cc#L589